src/components/SpiceList.js

```diff
@@ -2,13 +2,43 @@ import React from 'react'
 import SpiceItem from './SpiceItem'
 
 class SpiceList extends React.Component {
+
   state = {
     fourStarOnly: false,
     search: ""
   }
 
-  renderSpices() {
-    return this.props.spices.map(spice => (
+  handleChange = () => {
+    this.setState((prevState) => ({
+      fourStarOnly: !prevState.fourStarOnly
+    }))
+  }
+
+  onChange = (evt) => {
+    let searchTerm = evt.target.value.toLowerCase()
+    this.setState((prevState) => ({
+      search: searchTerm
+    }))
```

Comment:
There's nothing wrong with using the callback version of `setState` here, but it might be simpler to use the object version (since the next state doesn't depend on the previous one):

```js
    this.setState({
      search: searchTerm
    })
```